includes/woopay/class-woopay-session.php

```diff
@@ -535,7 +534,9 @@ private static function has_valid_request_signature() {
 	 * @return bool True if WooPay is enabled, false otherwise.
 	 */
 	private static function is_woopay_enabled(): bool {
-		return WC_Payments_Features::is_woopay_eligible() && 'yes' === WC_Payments::get_gateway()->get_option( 'platform_checkout', 'no' );
+		$woopay_enabled = get_option( 'woocommerce_woocommerce_payments_settings', [] )['platform_checkout'] ?? 'no';
+
+		return self::is_woopay_eligible() && 'yes' === $woopay_enabled;
```

Comment:
Since we delayed the initialization, can we go back to relying on the `WC_Payments_Features` and `WC_Payments` classes here? Or should we still decouple the code this way? 🤔 

I'm _leaning_ towards going back to using the WooPayments classes instead of decoupling here since we have guaranteed the other classes have been initialized.

Reply:
Yeah, if this is testing well with the delayed initialization, I would want to go back to the original implementation here to avoid some duplicated code.